### ISSUE TITLE
Experience flow should match later gens

### ISSUE THREAD

**Issue by DizzyEggg** [DizzyEggg/pokeemerald#344]
Lunos said:

My Torchic with Life Orb killed an opponent Poochyena in a double battle, it died afterward because of the Life Orb, and the exp. obtained from killing that Poochyena wasn't distributed until their ally used a move.
https://streamable.com/0voys
I don't recall how double battles work specially in the newer games, but shouldn't the exp. be distributed right after the kill?

**Comment by LOuroboros** [comment]
This matter was more or less discussed a little bit on Discord, so I'll proceed to bring up the most relevant part.
I did test double battles in Pokémon Ultra Sun not so long ago, and I was able to confirm that the current behavior of "fainted by Life Orb" in the battle_engine_v2 is not accurate.
https://streamable.com/uxgm9y

The opponent's ally is supposed to use their move after the exp. from having killed the first opponent is being given to you, not before.
